service/src/main/java/uk/gov/hmcts/reform/fpl/service/UserService.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.config.LocalAuthorityLookupConfiguration;
+import uk.gov.hmcts.reform.idam.client.IdamApi;
+import uk.gov.hmcts.reform.idam.client.models.UserDetails;
+
+import java.util.Map;
+
+/**
+ * Gets Users from Idam.
+ */
+@Service
+public class UserService {
+
+    private final IdamApi idamApi;
+    private final LocalAuthorityLookupConfiguration localAuthorityLookupConfiguration;
+
+    @Autowired
+    public UserService(IdamApi idamApi, LocalAuthorityLookupConfiguration localAuthorityLookupConfiguration) {
+        this.idamApi = idamApi;
+        this.localAuthorityLookupConfiguration = localAuthorityLookupConfiguration;
+    }
+
+    /**
+     * Makes a request to Idam and returns the user details for the user who is making the request.
+     *
+     * @param authorization the authorization token generated by the callback.
+     * @return caseLocalAuthority gained from the extractEmailDomain function.
```

Comment:
I do not think we should say what internal method is used. It will change and from user who makes call should not care. What is important is what return value is.